File: app/src/main/java/com/huynd/skyobserver/models/cheapestflight/CheapestPricePerMonthBody.java
package com.huynd.skyobserver.models.cheapestflight;

import static com.huynd.skyobserver.utils.Constants.CARRIERS;

/**
 * Created by HuyND on 11/18/2017.
 */
public class CheapestPricePerMonthBody {
    String startDate;
    String endDate;
    int minPrice;
    int maxPrice;
    String[] providers;
    String[] routes;
    String type;

    public CheapestPricePerMonthBody(String startDate, String endDate, String[] routes) {
        this.startDate = startDate;
        this.endDate = endDate;
        minPrice = 0;
        maxPrice = 700000;
        providers = CARRIERS;
        this.routes = routes;
        type = "month";
    }
}

File: app/src/main/java/com/huynd/skyobserver/models/cheapestflight/CheapestPricePerMonthBody.kt
package com.huynd.skyobserver.models.cheapestflight

import com.huynd.skyobserver.utils.Constants.Companion.CARRIERS

/**
 * Created by HuyND on 11/18/2017.
 */
class CheapestPricePerMonthBody(strYear: String, strMonth: String, strDay: String) {
    var startDate: String = ""
    var endDate: String = ""
    val minPrice = 0
    val maxPrice = 900000
    val providers = CARRIERS.asList()
    val routes: MutableList<String> = mutableListOf()
    val type = "date"

    init {
        startDate = "$strYear$strMonth$strDay"
        endDate = startDate
    }

    fun setRoutes(routes: List<String>) {
        this.routes.clear()
        this.routes.addAll(routes)
    }
}
